### ISSUE TITLE
Image.getSize does not use headers

### ISSUE THREAD

**Issue by jberns88** [archriss/react-native-image-gallery#74]
When loading images with something like:
``` JSON
[
  {
    "source": {
      "uri": "http://my.files.com/123",
      "headers": {
        "Authorization": "BASIC ..."
      }
    }
  }
]
```
The gallery errors because the getSize cannot be given headers and therefore sets the state to error. 
As images are cached by React Native once they're loaded is it possible to run the getSize method after the image is loaded as a workaround?

**Comment by shubham1164** [comment]
Hi @jberns88, I am facing a similar issue, have you found the answer on how to send headers while using Image.getSize(). I need this because I can't access the image without auth token that must be present in the header. Thanks in advance

**Comment by jberns88** [comment]
Nope, I ended up fetching the image myself and passing it in as base64. It's not very efficient but did the job.

**Comment by shubham1164** [comment]
Ammm....bad luck for me then...Thanks

**Comment by Quintis1212** [comment]
@jberns88 , @shubham1164 Hi ! There are already get size method  with headers https://reactnative.dev/docs/image#getsizewithheaders
